internal/controller/pod_event_handler.go

```diff
@@ -188,13 +194,25 @@ func (h *PodEventHandler) handlePodDeleted(ctx context.Context, pod *corev1.Pod,
 	logger := logf.FromContext(ctx).WithValues("pod", pod.Name, "workspace", workspaceName)
 	logger.Info("Workspace pod has been deleted", "podUID", pod.UID)
 
-	// Clean up SSM managed nodes for this pod
-	if h.ssmRemoteAccessStrategy == nil {
-		logger.Error(nil, "SSM remote access strategy not available - cannot cleanup SSM managed nodes")
-	} else {
-		if err := h.ssmRemoteAccessStrategy.CleanupSSMManagedNodes(ctx, pod); err != nil {
-			logger.Error(err, "Failed to cleanup SSM managed nodes")
+	// Check if this pod uses SSM remote access strategy by checking labels
+	accessStrategyName := pod.Labels[LabelAccessStrategyName]
```

Comment:
sanity-check: the pod has a label w/ the access strategy?

consider adding inline comment to explain how this label get propagated all the way to the pod.